Pidaq/Lib/database/database.py

```diff
@@ -0,0 +1,194 @@
+
+    Attributes:
+        client (InfluxDBClient): Instance of InfluxDB api to interact with the
+                                 database
+        current_database (str): Current database that the class is interacting
+                                with
+    """
+
+    def __init__(self, database: str) -> None:
+        """
+
+        Args:
+            database (str): Name of database to connect to. If the database
+                            does not exist, it will be created.
+        """
+
+        self.client = InfluxDBClient(host='localhost', port=8086)
+        self.current_database = database
+        current_databases = self.client.get_list_database()
+        if not any(current_database['name'] == database
+                   for current_database in current_databases):
+            self.client.create_database(database)
+        self.client.switch_database(database)
+
+    def switch_database(self, database: str) -> None:
+        """Switches currently active database"""
+
+        current_databases = self.client.get_list_database()
+        if not any(self.current_database['name'] == database
+                   for self.current_database in current_databases):
+            self.client.create_database(database)
+        self.client.switch_database(database)
+        self.current_database = database
+
+    def log_data(self, data: dict, measurement: str, tags: dict) -> None:
+        """Logs a single data point to the database"""
+
+        table_row = [{
+            'measurement': measurement,
+            'time': datetime.datetime.now(),
+            'fields': data,
+        }]
+        self.client.write_points(table_row, tags=tags)
+
+    def read_data(self, tags=None, measurements=None) -> str:
+        """Reads database for specified measurements, including specified tags and fields
+
+        Args:
+            tags (list(str)): Contains the tags and fields to be included in the query
+            measurements (list(str)): Contains the measurements to be queried
+
+        Returns:
+            (str) Raw json data
+        """
+
+        if tags is not None and measurements is None:
+            formatted_tags = format_data(tags)
+            data = self.client.query(
+                f'SELECT {formatted_tags}',
+                database=self.current_database
+            )
+        elif tags is None and measurements is not None:
+            formatted_measurements = format_data(measurements)
+            data = self.client.query(
+                f'SELECT * FROM {formatted_measurements}',
+                database=self.current_database
+            )
+        elif tags is not None and measurements is not None:
+            formatted_tags = format_data(tags)
+            formatted_measurements = format_data(measurements)
+            data = self.client.query(
+                f'SELECT {formatted_tags} FROM {formatted_measurements}',
+                database=self.current_database
+            )
```

Comment:
This function is a little bit different then I had intended. Ideally, the user can choose a tag value (i.e. profile="nicks-profile"), a measurement, and a set of fields. The function then returns all values of the selected fields that have the profile value = "nicks-profile". It is like filtering for a specific profile. If we use this design we could then have tags profileName, and testNo, so we can have numerous test runs using the same profile all in the same table and just filter for the desired ones. My idea for a tag was like a filter as opposed to a field.

The query would be something like: `"SELECT * FROM {MEASUREMENT} WHERE {tagname}=={tagvalue}"`, where tag is a dict like `{"name": "value"}`

HOWEVER, this is unnecessary if we decide to clear the tables after every test. In that case, there will only be data from one test profile at a time in the table. In this case, I still think the "tags" parameter of this function would be better named "fields".

We can call and go through this quickly to clear it up - finding it a little hard to verbalize rn